Automattic/pocket-casts-android
# Replace search menu with search bar

## Description
This replaces the search menu with a search bar on the podcasts screen.

(Part of up next in tab bar and profile tab changes)

## Testing Instructions
1. Start with no subscribed podcasts
2. Open podcasts tab 
3. ✅ Notice that you see a search bar and the search menu is removed
4. Subscribe to a few podcasts (at least 15 in count) 
5. ✅ Notice that search bar is still shown
6. Scroll to the end of the list and then slightly scroll to the top
7. ✅ Notice that search bar is hidden on scrolling towards bottom of the list and reappears on scroll to the top (you do not have to scroll to the top of the list to see the search bar)
8. Create a folder and add a few podcasts to it
9. Tap on folder
10. ✅ Notice that search bar is not shown
11. Tap on search bar
12. ✅ Notice that full-screen search screen is shown
13. ✅ Notice that you still see `search_shown` track event on tapping search bar
14. ✅  Notice that adding search bar has not caused any regression to pull-to-refresh or podcast rearrange behavior

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/1405144/aab3036f-3022-48ab-9599-c4a90a267d22

Blank screen

Dark theme | Light theme
----|----
<img width=200 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/524f57f7-f0c9-4dfe-8038-8d3216fc01ec"/> | <img width=200 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/5c9d1c05-aacf-4b29-8fc3-205062febad3"/>


## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md - To be added
 
#### I have tested any UI changes...
